flink-filesystems/flink-azure-fs-hadoop/src/main/java/org/apache/flink/fs/azurefs/AbstractAzureFSFactory.java

```diff
@@ -75,18 +76,27 @@ public void configure(Configuration config) {
     @Override
     public FileSystem create(URI fsUri) throws IOException {
         checkNotNull(fsUri, "passed file system URI object should not be null");
-        LOG.info("Trying to load and instantiate Azure File System");
+        LOG.info("Trying to load and instantiate Azure File System for {}", fsUri);
         return new HadoopFileSystem(createInitializedAzureFS(fsUri, flinkConfig));
     }
 
-    // uri is of the form: wasb(s)://yourcontainer@youraccount.blob.core.windows.net/testDir
+    // uri is of the form: wasb(s)://yourcontainer@youraccount.blob.core.windows.net/testDir (or)
+    // abfs(s):////yourcontainer@youraccount.dfs.core.windows.net/testDir
     private org.apache.hadoop.fs.FileSystem createInitializedAzureFS(
             URI fsUri, Configuration flinkConfig) throws IOException {
         org.apache.hadoop.conf.Configuration hadoopConfig = configLoader.getOrLoadHadoopConfig();
-
-        org.apache.hadoop.fs.FileSystem azureFS = new NativeAzureFileSystem();
-        azureFS.initialize(fsUri, hadoopConfig);
-
-        return azureFS;
+        String scheme = fsUri.getScheme();
+
+        if (scheme.startsWith("wasb")) {
+            LOG.info("Trying to initialize hadoop filesystem for {}.", scheme);
+            org.apache.hadoop.fs.FileSystem azureFS = new NativeAzureFileSystem();
+            azureFS.initialize(fsUri, hadoopConfig);
+            return azureFS;
+        } else {
```

Comment:
Do we need to check if the scheme prefix starts with "abfs" here to cover both the secure and none secure schemes without including other schemes that are non Azure file systems?

Reply:
Not necessarily, But anyway this code is going to get refactored and this if else condition will go away.